File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Gui/App/Models/Dossiers/DossierModel.ts
import DossierRoleModel = require('App/Models/Dossiers/DossierRoleModel');
import DossierDocumentModel = require('App/Models/Dossiers/DossierDocumentModel');
import DossierFolderModel = require('App/Models/Dossiers/DossierFolderModel');
import ContainerModel = require('App/Models/Commons/ContainerModel');
import ContactModel = require('App/Models/Commons/ContactModel');
import MetadataRepositoryModel = require('App/Models/Commons/MetadataRepositoryModel');
import CategoryModel = require('App/Models/Commons/CategoryModel');
import DossierType = require('App/Models/Dossiers/DossierType');
import DossierStatus = require('App/Models/Dossiers/DossierStatus');

interface DossierModel {
    UniqueId: string;
    Year: number;
    Number: string;
    Subject: string;
    Note: string;
    Container: ContainerModel;
    StartDate: Date;
    EndDate?: Date;
    MetadataDesigner: string;
    MetadataValues: string;
    RegistrationDate: Date;
    RegistrationUser: string;
    LastChangedUser: string;
    LastChangedDate: Date;
    MetadataRepository: MetadataRepositoryModel;
    DossierRoles: DossierRoleModel[];
    Contacts: ContactModel[];
    DossierDocuments: DossierDocumentModel[];
    DossierFolders: DossierFolderModel[];
    Category: CategoryModel;
    DossierType: DossierType;
    Status: DossierStatus;
}

export = DossierModel;

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Model/Entities/Dossiers/DossierModel.cs
using System;
using System.Collections.Generic;
using VecompSoftware.Commons.Interfaces.CQRS.Commands;
using VecompSoftware.DocSuiteWeb.Model.Entities.Commons;

namespace VecompSoftware.DocSuiteWeb.Model.Entities.Dossiers
{
    public class DossierModel : IContentBase
    {
        #region [ Constructor ]
        public DossierModel()
        {
            Roles = new HashSet<DossierRoleModel>();
            Contacts = new HashSet<ContactModel>();
            Documents = new HashSet<DossierDocumentModel>();
            DossierFolders = new HashSet<DossierFolderModel>();
        }
        #endregion

        #region [ Proprieties ]
        public Guid UniqueId { get; set; }

        public short Year { get; set; }

        public int Number { get; set; }

        public string Title { get; set; }

        public string Subject { get; set; }

        public string Note { get; set; }

        public DateTimeOffset RegistrationDate { get; set; }

        public string RegistrationUser { get; set; }

        public DateTimeOffset? LastChangedDate { get; set; }

        public string LastChangedUser { get; set; }

        public DateTimeOffset StartDate { get; set; }

        public DateTimeOffset? EndDate { get; set; }

        public short ContainerId { get; set; }

        public string ContainerName { get; set; }

        public string MetadataDesigner { get; set; }

        public string MetadataValues { get; set; }

        public CategoryModel Category { get; set; }

        public DossierType DossierType { get; set; }

        public DossierStatus Status { get; set; }
        #endregion

        #region [ Navigation Proprieties ]
        public MetadataRepositoryModel MetadataRepository { get; set; }

        public ICollection<DossierRoleModel> Roles { get; set; }

        public ICollection<ContactModel> Contacts { get; set; }

        public ICollection<DossierDocumentModel> Documents { get; set; }

        public ICollection<DossierFolderModel> DossierFolders { get; set; }
        #endregion
    }
}
